project-koku/koku
# [COST-4973] - Fix mid month finalization

## Jira Ticket

[COST-4973](https://issues.redhat.com/browse/COST-4973)

## Description

This change will add an additional check to prevent us from potentially running full month reprocessing/finalisation on current months. This also prevents us from deleting ALL parquet files by mistake. 

## Testing

1. Checkout Branch
2. Restart Koku with multiple workers ideally
3. Add AWS source
4. Run initial ingest!
5. Generate some new nise data (This needs to be split files, --file-row-limit)
6. Hit download
7. Verify only 5 days of parquet files are deleted and recreated

Feel free to compare to main, same steps as above but all parquet files outside the processing range will get cleared by mistake.

## Release Notes
- [ ] Correctly handle AWS end of Tax year billing reports

```markdown
* [COST-4973](https://issues.redhat.com/browse/COST-4973) Correctly handle AWS end of Tax year billing reports
```
